docs/docs/for-developers/building.md

```diff
@@ -11,6 +11,8 @@ sidebar_label: Building from Source
 - Install [Git](https://git-scm.com/)
 - Install [Node](https://nodejs.org/en)
 - Install [Esy](https://esy.sh) (__0.6.2__ or above is required, but the latest version is recommened: `npm install -g esy@latest`)
+> __NOTE:__ **Linux-only**: if you need to install using `sudo npm -g esy@latest` then you might run into this issue [esy/esy#1099](https://github.com/esy/esy/issues/1099), the actual workaround is the following :`sudo npm install -g --unsafe-perm esy`.
+
```

Comment:
If you have to use `sudo` then your npm installation is broken<sup>1</sup>. The proper solution is to fix your installation according to [the official documentation](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally) or [this tep-by-step guide](https://github.com/sindresorhus/guides/blob/master/npm-global-without-sudo.md), not to invite the entire internet to a party with full access to your computer. The flag is called "unsafe" for good reason.

<sup>1</sup> There are historical reasons why this is still so prevalent: NPM wasn't originally designed for installing packages globally, and most Linux package managers don't like other package managers meddling in their business.

Reply:
I see I understand, the thing is the unsafe flag being used here is only needed for the `esy` package other packages installations works fine without that additionnal flag but since my NPM install is kinda broken  I guess these changes are useless.

Sorry for having made this PR I should've known better. I understand that this is related to security concerns.

Reply:
No need to apologize. There's unfortunately a lot of material on this that's pointing in the wrong direction, and not a lot pointing in the right direction.

The problem is the use of `sudo` in the first place, not so much the use of `--unsafe-perm`. As I understand it, `--unsafe-perm` was added to try to mitigate some of the harm of using `sudo` by trying to install as the user invoking `npm` instead of the root user. Unfortunately that doesn't always work, since the boundaries are so fuzzy, and completely unenforced, some `postinstall` go outside them. Therefore, to enable the legitimate use case of installing npm packages globally for all users on a machine (as opposed to globally for _one_ user, which is usually the case), this flag was added. It's a real mess, for historical reasons as I said.

In any case, it's not my place to say what's acceptable or not around here, but I would at least avoid offering insecure advice directly. I think it's OK to link to the issue on the `esy` repo though, and if the `esy` developers want to offer insecure advice then that's on them. I've warned them about it before, and I do think Manas offers good advice there, though it's less convenient than the bad advice in the short term and what I'm advising here in the long-term, without offering much explanation of why it's worth it.

Reply:
I've updated the documentation note with a more accurate description of what should be done (I suppose).

Can you confirm to me that this looks good to you as of now ?

Reply:
Nitpicking here, but it's a bit wordy. I'd cut the last part after pointing to the esy issue. The recommendation given earlier should be sufficient. Also, technically it might not be broken. As I said earlier there are legitimate uses cases for installing packages globally for all users, where `sudo` and `--unsafe-perm` are needed, but then the admin certainly ought to know what they're doing. For single-user systems I'd consider it broken.

Reply:
It's fine to nitpick, no problem I understand that they are some grey cases where it may be okay to allow someone to do this when they know what they are doing, I updated the documentation following your comment and also said that the NPM install might be broken since this probably doesn't always hold true I guess.